### PR TITLE
Add support for SingleVariable objective

### LINKED ISSUE #49: Linear / quadratic models
If a JuMP model `m` has linear constraints +  linear or quadratic objective, we cannot use `NLPEvalaluator(m).constraints` because it's an undefined reference. We can detect if we have non linear objective with `eval.has_nlobj` but I don't know if we have something similar for non linear constraints...

Edit :`num_nl_constraints(m)` works.

### PR DESCRIPTION
**Pull request by amontoison**
close #49 

Objective like `min(t)` or `max(x[i])` are `SingleVariable` objective and not `ScalarAffineFunction` or 
`ScalarQuadraticFunction`.

### CODE REVIEW

File: test/nlp_problems/linsv.jl [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+  @variable(nlp, x[i=1:2])
+
+  @constraint(nlp, x[1] + x[2] ≥ 3)
+  @constraint(nlp, x[2] - 1 ≥ 0)
+
+  @objective(nlp, Min, x[1])
+
+  return nlp
+end
```
Comment (abelsiqueira) [inline test/nlp_problems/linsv.jl:13]: Can you create an `autodiff` problem and run the consistency checks?

Comment (amontoison) [reply test/nlp_problems/linsv.jl:13]: Yes -> https://github.com/JuliaSmoothOptimizers/NLPModels.jl/pull/283

Comment (abelsiqueira) [reply test/nlp_problems/linsv.jl:13]: https://github.com/JuliaRegistries/General/pull/13978

**Comment by abelsiqueira** [comment]
Thanks

**Pull request merged**